src/cdomain/value/cdomains/threadIdDomain.ml

```diff
@@ -86,7 +86,7 @@ struct
     | ({vname; _}, None) -> List.mem vname @@ GobConfig.get_string_list "mainfun"
     | _ -> false
 
-  let is_unique _ = false (* TODO: should this consider main unique? *)
+  let is_unique = is_main
```

Comment:
Why is this sound? What if multiple threads are created from the main function is created multiple times?

Reply:
Oh, good point!

I now added an extra condition to the `is_main` function based on `ana.thread.include-node`. If nodes are included, then only the actual root thread has `None` (others have a node where they're created, even if it's the same `main` function). if nodes are not included, then we cannot tell them apart indeed and it should be non-unique.